snowflake_data_profiler/profiling/profiler.py

```diff
@@ -69,15 +84,19 @@ def get_profile_results(data):
              "cramers": {"calculate": False},
          },
         )
+
         p = profile.to_html() # this step sometimes fails with matplotlib errors about threads. I've only fixed it by adjusting requirements.txt in the past. I've just specified the specific versions of libraries. Pyarrow seems to have an impact on this.
         return p
+
     else:
         raise TypeError('This is not a pandas dataframe.')
 
 
-def do_profile():
-    pd_df = connect_to_snowflake()
+def get_profile(sfUser, sfPswd, sfURL, sfDatabase, sfSchema, sfTable, sfRole=None, sfWarehouse=None):
+    """main function"""
+
+    conn = get_snowflake_connection(sfUser, sfPswd, sfURL, sfDatabase, sfSchema, sfTable, sfRole)
+    pd_df = get_pandas_dataframe(conn, sfDatabase, sfSchema, sfTable, sfWarehouse)
     return get_profile_results(pd_df)
```

Comment:
for debugging purposes, it's usually a good idea to set this equal to a variable value that we can inspect if we have to. Returns like this can be common in memory-constrained environments, like with embedded development, but they're not necessary here.